ggshield/iac/iac_scan_models.py

```diff
@@ -33,40 +43,86 @@ class IaCDiffScanResult(Base):
+    and the content type is JSON.
+    """
+    return (
+        resp.headers["content-type"] == "application/json" and resp.status_code == 200
+    )
+
+
+# TODO: Delete this after moving iac_diff_scan into pygitguardian
+def load_detail(resp: Response) -> Detail:
+    """
+    load_detail loads a Detail from a response
+    be it JSON or html.
+
+    :param resp: API response
+    :type resp: Response
+    :return: detail object of response
+    :rtype: Detail
+    """
+    if resp.headers["content-type"] == "application/json":
+        data = resp.json()
+    else:
+        data = {"detail": resp.text}
 
-class MockClient(GGClient):
-    # TODO: move this into GGClient
-    def mock_api_iac_diff_scan(
+    return cast(Detail, Detail.SCHEMA.load(data))
+
+
+# TODO: delete this class after updating pygitguardian
+class TmpGGClient(GGClient):
+    """Temporary class used until iac_diff_scan can be moved to Ggclient."""
+
+    # TODO: Move this method into GGClient.
+    def iac_diff_scan(
         self,
         reference: bytes,
         current: bytes,
         scan_parameters: IaCScanParameters,
         extra_headers: Optional[Dict[str, str]] = None,
     ) -> Union[Detail, IaCDiffScanResult]:
-        scan = self.iac_directory_scan(
-            Path("."),
-            [],
-            scan_parameters,
-            extra_headers,
-        )
-        if isinstance(scan, Detail):
-            return scan
-        result = IaCDiffScanResult(
-            scan.id,
-            scan.type,
-            scan.iac_engine_version,
-            entities_with_incidents=IaCDiffScanEntities([], [], []),
-        )
-        result.status_code = 200
+        result: Union[Detail, IaCDiffScanResult]
+        try:
+            resp = self.post(
+                endpoint="iac_diff_scan",
+                extra_headers=extra_headers,
+                files={
+                    "reference": reference,
+                    "current": current,
+                },
+                data={
+                    "scan_parameters": IaCScanParametersSchema().dumps(scan_parameters),
+                },
+            )
+        except requests.exceptions.ReadTimeout:
+            result = Detail("The request timed out.")
+            result.status_code = 504
+        else:
+            if is_ok(resp):
+                # IaCDiffScanResultSchema.from_dict(resp.json()) does not work for now
+                # TODO: replace load with from_dict once this is moved into pygitguardian
+                result = IaCDiffScanResultSchema().load(resp.json())  # type: ignore
+            else:
+                result = load_detail(resp)
+
+            result.status_code = resp.status_code
         return result
 
 
-# TODO: remove this once api_iac_diff_scan is moved into GGClient
-def create_mock_client_from_config(config: Config) -> MockClient:
+# TODO: remove this once iac_diff_scan is moved into GGClient
+def create_mock_client_from_config(config: Config) -> TmpGGClient:
```

Comment:
Do not call this "mock": "mock" is a test-domain word. Since it's in a separate module from core.client, you can name it `create_client_from_config()`. We should never import both methods in the same file so it should not cause any conflict.